Tasks/SonarQubePostTest/PRCA/Orchestrator.ps1

```diff
@@ -1,18 +1,113 @@
+    Assert ( $vssConnection -ne $null ) "Internal error: could not retrieve the VssConnection object"
+        
+    $gitClient = $vssConnection.GetClient("Microsoft.TeamFoundation.SourceControl.WebApi.GitHttpClient")            
+    $discussionClient = $vssConnection.GetClient("Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionHttpClient")    
+    $codeReviewClient = $vssConnection.GetClient("Microsoft.VisualStudio.Services.CodeReview.WebApi.CodeReviewHttpClient") # ???
+        
+    Assert ( $gitClient -ne $null ) "Internal error: could not retrieve the GitHttpClient object"
+    Assert ( $discussionClient -ne $null ) "Internal error: could not retrieve the DiscussionHttpClient object"
+    Assert ( $codeReviewClient -ne $null ) "Internal error: could not retrieve the CodeReviewHttpClient object"
+             
+    $sourceBranch =  GetTaskContextVariable "Build.SourceBranch"
+    $pullRequestId = RetrievePullRequestId $sourceBranch
+    Write-Verbose  "Pull Request Id $pullRequestId"
+    
+    $repositoryId = GetTaskContextVariable "build.repository.id"    
+    $teamProject = GetTaskContextVariable "system.teamProject"
+
+    Assert (![String]::IsNullOrWhiteSpace($repositoryId)) "Internal error: could not determine the build.repository.id"
+    Assert (![String]::IsNullOrWhiteSpace($teamProject)) "Internal error: could not determine the system.teamProject"   
+
+    $pullRequest = $gitClient.GetPullRequestAsync($project, $repositoryId, $pullRequestId).Result;
+    $artifactUri = GetArtifactUri $pullRequest.CodeReviewId $pullRequest.Repository.ProjectReference.Id $pullRequestId
+           
+    ProcessIssues $newIssues $codeReviewClient $gitClient $discussionClient $pullRequest $teamProject $repositoryId $sourceBranch $artifactUri        
+}
+
+function LoadClientOMAssemblies
+{                    
+    Write-Verbose "Loading TFS client object model assemblies packaged with the build agent"
+    
+    $tfsClientOMDir = $env:AGENT_SERVEROMDIRECTORY
+    
+    $externalAssemblyNames = (             
+        "Microsoft.TeamFoundation.Common.dll",
+        "Microsoft.TeamFoundation.Core.WebApi.dll",
+        "Microsoft.TeamFoundation.SourceControl.WebApi.dll",             
+        "Microsoft.VisualStudio.Services.CodeReview.Common.dll",
+        "Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.dll",    
+        "Microsoft.VisualStudio.Services.CodeReview.WebApi.dll",        
+        "Microsoft.VisualStudio.Services.Common.dll",
+        "Microsoft.VisualStudio.Services.WebApi.dll",
+        "Newtonsoft.Json.dll")
+       
+    $externalAssemblyPaths = $externalAssemblyNames | foreach { [System.IO.Path]::Combine($tfsClientOMDir, $_)}                        
+    $externalAssemblyPaths | foreach {Add-Type -Path $_}    
+    
+    Write-Verbose "Loaded $externalAssemblyPaths"
+}
+
+function RetrievePullRequestId 
```

Comment:
No, there is no UI ☺. Assert is a function I wrote that simply throws if that condition fails.

Bogdan

From: Eyal Lantzman [mailto:notifications@github.com]
Sent: Tuesday, March 1, 2016 3:32 PM
To: Microsoft/vso-agent-tasks <vso-agent-tasks@noreply.github.com>
Cc: Bogdan Gavril <bgavril@microsoft.com>
Subject: Re: [vso-agent-tasks] Have the orchestrator new up all the actors needed to publish issues (#1319)


In Tasks/SonarQubePostTest/PRCA/Orchestrator.ps1<https://github.com/Microsoft/vso-agent-tasks/pull/1319#discussion_r54583078>:

> +        "Microsoft.TeamFoundation.Core.WebApi.dll",

> +        "Microsoft.TeamFoundation.SourceControl.WebApi.dll",

> +        "Microsoft.VisualStudio.Services.CodeReview.Common.dll",

> +        "Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.dll",

> +        "Microsoft.VisualStudio.Services.CodeReview.WebApi.dll",

> +        "Microsoft.VisualStudio.Services.Common.dll",

> +        "Microsoft.VisualStudio.Services.WebApi.dll",

> +        "Newtonsoft.Json.dll")

> +

> +    $externalAssemblyPaths = $externalAssemblyNames | foreach { [System.IO.Path]::Combine($tfsClientOMDir, $_)}

> +    $externalAssemblyPaths | foreach {Add-Type -Path $_}

> +

> +    Write-Verbose "Loaded $externalAssemblyPaths"

> +}

> +

> +function RetrievePullRequestId

Shouldn't we throw in some of those cases? What will happen to the server when you assert - will it be a blocking window that you need to attach a debugger to it (in your remote agent) ?

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/vso-agent-tasks/pull/1319/files#r54583078>.